build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/JdkDownloadPluginFuncTest.groovy

```diff
@@ -99,17 +99,13 @@ class JdkDownloadPluginFuncTest extends AbstractGradleFuncTest {
         given:
         def mockRepoUrl = urlPath(jdkVendor, jdkVersion, platform)
         def mockedContent = filebytes(jdkVendor, platform)
-        3.times {
-            settingsFile << """
-                include ':sub-$it'
-            """
-        }
         buildFile.text = """
             plugins {
              id 'elasticsearch.jdk-download' apply false
             }
-
-            subprojects {
+        """
+        3.times {
+            subProject(':sub-' + it) << """
```

Comment:
I had issues with these tests failing due to tasks not being found. It appeared to have to do with using the subproject notation and filtering to the "sub-" prefix (otherwise it picks up the native libs project). So I made the test setup the subproject build files directly instead of configuring through the parent. I opened https://github.com/elastic/elasticsearch/pull/105998 to isolate these changes.